Plans: add current plan page

artpi opened the pull request:
# Don't be afraid of :+1: . This is hidden behind a feature flag.

This repurposes splendid "Thank you" page and shows it instead of /plans for plan owners.
This is intended to:
- make plan owners feel happy about their purchase
- explain how to use their brand new features
- reduce refunds

Props to @drewblaisdell @scruffian for making this components after the checkout flow

## UI

### Premium plan owners

<img width="376" alt="zrzut ekranu 2016-04-15 o 17 01 58" src="https://cloud.githubusercontent.com/assets/3775068/14567416/df3f7e88-032b-11e6-9705-01b925e8eae9.png">


### Business plan owners
<img width="378" alt="zrzut ekranu 2016-04-15 o 17 01 46" src="https://cloud.githubusercontent.com/assets/3775068/14567405/d7e2804a-032b-11e6-87e9-15f76023cd51.png">

## Testing
Go to
http://calypso.localhost:3000/plans/my-plan/planpremium.wordpress.com
replace with your site slug..

Link and ABtest for linking it from sidebar will be introduced in #4783 

CC
@rralian @mtias @gziolo @gwwar

mtias commented:
The big blue banner shouldn't be there :)

mtias commented:
or at least should be made much smaller for this context.

mtias commented on client/my-sites/plans/index.js:
```diff
@@ -31,6 +32,14 @@ export default function() {
+			retarget,
+			controller.siteSelection,
+			controller.navigation,
+			plansPageYourPlan
```
can this be just `yourPlan`?

artpi replied:
Yup!

mtias commented on client/my-sites/upgrades/checkout-thank-you/plans-page-your-plan.js:
```diff
@@ -0,0 +1,96 @@
+/**
```
We should place this in `my-sites/plan` or `my-sites/plans/current-plan`

artpi replied:
@mtias, I was thinking about that.
But what about CSS ?
I dont want to duplicate all of these classes, because I later want to improve this page and I kinda want to avoid huge refactors of yet another thing.

So, If I move it to `my-sites/plan` or something similar, will you let me keep current CSS ?
If not, Im not moving it anywhere :)

mtias replied:
What CSS do you need to reuse?

drw158 commented:
@mtias Maybe just a regular banner notice?

mtias commented:
@drw158 what do you have in mind?

drw158 commented:
@artpi Sorry, I haven't been following this one. Will this be displayed after a plan purchase, or is this just what they see when they click on Plan?

I'm worried about
- Having a back button so they can see the plans if they wanted to.
- A user seeing this several months after purchasing and they get confused.

artpi commented:
@drw158, all valid points!
> is this just what they see when they click on Plan

Yes!

> Having a back button so they can see the plans if they wanted to.

I dont think back button is good for this, but we need to link somehow to `/plans/compare`, for sure!

> A user seeing this several months after purchasing and they get confused.

The thank-you messaging has to be tweaked, as @mtias pointed :)

mtias commented:
Yeah, let's plan to have a `<Card href="/plans/compare">` at the bottom.

mtias commented on client/my-sites/plans/index.js:
```diff
@@ -31,6 +32,14 @@ export default function() {
 		);
 
 		page(
+			'/plans/my-plan/:site',
```
I think it'd be better to use `/plan` singular for this. Thoughts?

artpi replied:
I am not convinces, I like how it fits with `/plans/feature...` subtree. I like keeping consistent tree more than matching the plural/singular form

mtias commented on client/my-sites/plans/controller.jsx:
```diff
@@ -52,6 +55,12 @@ export default {
 
+		//Redirect for upgraded sites
+		if ( site && ( isPremium( site.plan ) || isBusiness( site.plan ) ) ) {
+			page.redirect( '/plans/my-plan/' + site.slug );
```
I think instead of redirecting we should just update the sidebar link for the "plans" item.

artpi replied:
done!

drw158 commented:
@artpi What if they only have Premium? How will they know there is a business plan? And how will they buy a business plan if they know one exists?

mtias commented on client/my-sites/upgrades/checkout-thank-you/plans-page-your-plan.js:
```diff
@@ -0,0 +1,96 @@
+		context.store.dispatch( fetchSitePlans( selectedSite.ID ) );
+	}
+
+	const PlanDetailsComponent = React.createClass( {
```
Also, let's just separate controllers from the components to be rendered. The general structure is:

```
- index.js (declares the routes as is the entry point for the whole section)
- controller.js (handles functions to be executed as routes are matched
- main.jsx (is the primary render tree for your area and includes the <Main> component)
```

mtias replied:
If we are reusing the index and controller from `/plans` then we only need to create the `main` file in `/plan/main.jsx`

artpi replied:
While you were writing this comment I was doing exactly that :D

artpi commented:
@drw158 : 
> What if they only have Premium? How will they know there is a business plan? And how will they buy a business plan if they know one exists?

Added few nudges and links, this is how it looks for premium:

<img width="1035" alt="zrzut ekranu 2016-04-15 o 11 02 26" src="https://cloud.githubusercontent.com/assets/3775068/14559444/ad923252-0300-11e6-846f-5d4fc3130a61.png">

mtias commented:
@artpi I'd only use the plans-compare component @gwwar created at the very bottom of the page. The nudge and centered card seem out of place.

mtias commented on client/my-sites/plans/current-plan/main.jsx:
```diff
@@ -0,0 +1,86 @@
+					{ this.translate( 'See all the amazing features in our various plans' ) }
+				</Card>
+
+				<Card className="checkout-thank-you__footer">
```
Should drop this class.

artpi replied:
I will, I will

artpi replied:
Done

artpi commented:
>  I'd only use the plans-compare component @gwwar created at the very bottom of the page.

I would like to A/B test that as a follow-up, because I strongly feel that introducing planscompare on this page will introduce too much information. Especially, that our bet is that this information is much more rarely needed.

I tweaked a lot of stuff, adressed comments and introduced A/B test. I would like to merge this :)
Description at the top has screenshots and updated info

mtias commented on client/my-sites/plans/index.js:
```diff
@@ -31,6 +32,14 @@ export default function() {
+			controller.siteSelection,
+			controller.navigation,
+			yourPlan
+		);
```
We should have a route for `my-plan` that uses `controller.sites`

drw158 commented:
1. Definitely should enable on wpcalypso first.
2. Text could use a lot of work.
3. Definitely shouldn't put a nudge component on this page. We should be restrained when it comes to nudges.
4. I like the idea of using the plan compare component.

mtias commented:
Update showing current status:

![image](https://cloud.githubusercontent.com/assets/548849/14567134/fcfcc514-0332-11e6-94f3-b1fe54726518.png)

artpi commented:
Edit:
## Added new screenshots in the descriptions at the top

> Definitely should enable on wpcalypso first.

Working on it

> Text could use a lot of work.

Part of it is the fact that 90% of it reuses components from `checkout-thank-you` and we need to be careful how we change that, but I totally agree

> Definitely shouldn't put a nudge component on this page. We should be restrained when it comes to nudges.

Removed few commit ago :)

> I like the idea of using the plan compare component.

I have a lot of concerns, so maybe we can add that / discuss in a different PR :)

mtias commented on assets/stylesheets/_components.scss:
```diff
@@ -200,6 +201,7 @@
 @import 'my-sites/plan-storage/style';
 @import 'my-sites/plans/style';
 @import 'my-sites/plans/plan-feature/style';
+@import 'my-sites/plans/current-plan/style';
```
we sneaked two of these :D

artpi replied:
Yeah, I was just commenting on it :D
Removing!

artpi replied:
Done!

mtias commented on client/my-sites/plans/current-plan/controller.js:
```diff
@@ -0,0 +1,36 @@
+ * Internal Dependencies
+ */
+import { renderWithReduxStore } from 'lib/react-helpers';
+
```
extra line

artpi replied:
Fixed!

artpi commented:
I added a feature flag hiding all of it.

mtias commented:
@artpi this is looking good to me. Do you mind rebasing a bit?

artpi commented:
>  Do you mind rebasing and squashing a bit?

I went full nuclear. One commit, kind sir.

scruffian commented:
I'm not sure about the use of the ⭐ icons here. We have specific iconography for Premium and Business; it would probably be better to be consistent. cc @breezyskies

scruffian commented:
I wonder  if the HE gravatar should be dropped below the title, so it aligns with the top of the text box.

<img width="651" alt="screen shot 2016-04-18 at 19 14 19" src="https://cloud.githubusercontent.com/assets/275961/14614619/d5f42c28-0599-11e6-9019-bbc26a70f5e2.png">

scruffian commented:
I see `TypeError: product is undefined` when I clear local storage and reload the page.

scruffian commented:
The placeholder state could probably be improved:
<img width="669" alt="screen shot 2016-04-18 at 19 16 47" src="https://cloud.githubusercontent.com/assets/275961/14614702/2e9ad476-059a-11e6-8395-cd5c5151bec1.png">

scruffian commented on client/my-sites/plans/current-plan/main.jsx:
```diff
@@ -0,0 +1,99 @@
+							</div>
+						</div>
+					</div>
+					<CheckoutThankYouFeaturesHeader
```
If this component is going to be used outside the context of checkout then we should probably move it to the components directory and give it a new name.

scruffian commented:
The mobile view could do with a few tweaks:
<img width="493" alt="screen shot 2016-04-18 at 19 27 42" src="https://cloud.githubusercontent.com/assets/275961/14615050/d0c5c002-059b-11e6-90ee-0484d777331c.png">

scruffian commented:
This is a great idea. Just needs a few tweaks I think :)

breezyskies commented:
> I'm not sure about the use of the ⭐ icons here. We have specific iconography for Premium and Business; it would probably be better to be consistent.

Agreed -- this should probably use the correct [plan icons](https://github.com/Automattic/wp-calypso/tree/master/public/images/plans) rather than the star.

On a related note, the ["free trial hub"](https://github.com/Automattic/wp-calypso/issues/1151#issuecomment-162624443), while only implemented for trials, was intended to have a similar purpose. Definitely worth looking at as you continue to iterate on this.

artpi commented:
@scruffian, thanks for taking a look!
> I'm not sure about the use of the ⭐ icons here

@adambbecker is introducing new plans messaging, where star generally means an upgrade, so it will be consistent.

> I see TypeError: product is undefined when I clear local storage and reload the page.

Good catch! Fixing right now

>  If this component is going to be used outside the context of checkout then we should probably move it to the components directory

I will create an issue for that

> The placeholder state could probably be improved

Another issue :)

> The mobile view could do with a few tweaks:

true!

adambbecker commented:
@artpi, @scruffian: My plan for using the star was mainly tailored for the nudge use case. Once we've gotten to a full page like this one that just displays the current plan, having the actual plan icon / figure I think would make the most sense. Something like this:

![screen shot 2016-04-19 at 9 12 26 am](https://cloud.githubusercontent.com/assets/1427136/14640301/f0e7f328-060e-11e6-8d88-a0804104c86d.png)

artpi commented:
- I rewrote how the data is handled to only use redux
- Added proper placeholders
- Fixed styling on mobile

## Placeholders

![zrzut ekranu 2016-04-19 o 16 02 59](https://cloud.githubusercontent.com/assets/3775068/14642909/9f77a90a-064c-11e6-97ae-109a6c12a132.png)

## Mobile

![zrzut ekranu 2016-04-19 o 16 27 14](https://cloud.githubusercontent.com/assets/3775068/14642912/a62e1806-064c-11e6-82f3-5cd5dd86b5a4.png)

gwwar commented:
After a squash+rebase, should be good to 🚢
